src/libraries/Native/Unix/System.Globalization.Native/pal_timeZoneInfo.c

```diff
@@ -80,3 +63,238 @@ int32_t GlobalizationNative_IanaIdToWindowsId(const UChar* ianaId, UChar* window
+    // Note that we leave the calendar open and close it later so we can also get the base offset.
+    UChar standardName[DISPLAY_NAME_LENGTH];
+    UCalendar* calendar = ucal_open(timeZoneId, -1, locale, UCAL_DEFAULT, err);
+    if (U_FAILURE(*err)) goto exit;
+
+    ucal_setMillis(calendar, timestamp, err);
+    if (U_FAILURE(*err)) goto exit;
+
+    ucal_getTimeZoneDisplayName(calendar, UCAL_STANDARD, locale, standardName, DISPLAY_NAME_LENGTH, err);
+    if (U_FAILURE(*err)) goto exit;
+
+    // See if the generic name is the same as the standard name.
+    if (u_strcmp(genericName, standardName) == 0)
+    {
+        // Get some details for later comparison.
+        const int32_t originalGenericNameActualLength = u_strlen(genericName);
+        const int32_t baseOffset = ucal_get(calendar, UCAL_ZONE_OFFSET, err);
+        if (U_FAILURE(*err)) goto exit;
+
+        // Allocate some additional strings for test values.
+        UChar testTimeZoneId[TZID_LENGTH];
+        UChar testDisplayName[DISPLAY_NAME_LENGTH];
+        UChar testDisplayName2[DISPLAY_NAME_LENGTH];
+
+        // Enumerate over all the time zones having the same base offset.
+        UEnumeration* pEnum = ucal_openTimeZoneIDEnumeration(UCAL_ZONE_TYPE_CANONICAL_LOCATION, NULL, &baseOffset, err);
+        if (U_FAILURE(*err)) goto exitEnum;
+
+        int count = uenum_count(pEnum, err);
+        if (U_FAILURE(*err)) goto exitEnum;
+
+        for (int i = 0; i < count; i++)
+        {
+            // Get a time zone id from the enumeration to test with.
+            int32_t testIdLength;
+            const char* testId = uenum_next(pEnum, &testIdLength, err);
+            if (U_FAILURE(*err)) goto exitEnum;
+            u_uastrcpy(testTimeZoneId, testId);
+
+            // Get the standard name from the test time zone.
+            GetTimeZoneDisplayName_FromCalendar(locale, testTimeZoneId, timestamp, UCAL_STANDARD, testDisplayName, DISPLAY_NAME_LENGTH, err);
+            if (U_FAILURE(*err)) goto exitEnum;
+
+            // See if the test time zone has a different standard name.
+            if (u_strcmp(testDisplayName, standardName) != 0)
+            {
+                // It has a different standard name. We can't use it.
+                continue;
+            }
+
+            // Get the generic name from the test time zone.
+            GetTimeZoneDisplayName_FromPattern(locale, testTimeZoneId, timestamp, GENERIC_PATTERN_UCHAR, testDisplayName, DISPLAY_NAME_LENGTH, err);
+            if (U_FAILURE(*err)) goto exitEnum;
+
+            // See if the test time zone has a longer (or same size) generic name.
+            if (u_strlen(testDisplayName) >= originalGenericNameActualLength)
+            {
+                // The test time zone's generic name isn't any shorter than the one we already have.
+                continue;
+            }
+
+            // We probably have found a better generic name.  But just to be safe, make sure the test zone isn't
```

Comment:
>  [](start = 60, length = 1)

nit: remove the space.

Reply:
Do you mean the two spaces between the sentences?  I was educated on APA style. 😄

I didn't see anything on this in the project's style guides.  Unless it's mandated, I'd prefer to leave as is.

Reply:
> Do you mean the two spaces between the sentences? 

Yes 

> I was educated on APA style. 

Look at most of the code in our repo and will see we don't use double spaces. 

> I didn't see anything on this in the project's style guides. Unless it's mandated, I'd prefer to leave as is.

I am fine to leave it but expect some others will change it when touching this code again :-)

Reply:
Hmmm..  Regex search for `\S\s\s\S` shows a very lot of existing comments that have two spaces, in `TimeZoneInfo.cs`, `TimeZoneInfo.Unix.cs`, etc.  I should change them all - even those I didn't write?

Reply:
Going to leave the two spaces for comments as-is.